backend/amt.pm

```diff
@@ -95,19 +96,11 @@ sub set_power_state ($self, $power_state) {
 }
 
 sub select_next_boot ($self, $bootdev) {
-    my $amt_bootdev;
-    if ($bootdev eq 'cddvd') {
-        $amt_bootdev = 'Intel(r) AMT: Force CD/DVD Boot';
-    }
-    elsif ($bootdev eq 'hdd') {
-        $amt_bootdev = 'Intel(r) AMT: Force Hard-drive Boot';
-    }
-    elsif ($bootdev eq 'pxe') {
-        $amt_bootdev = 'Intel(r) AMT: Force PXE Boot';
-    }
-    else {
-        die "Unsupported boot device $bootdev";
-    }
+    my $amt_bootdev = 'Intel(r) AMT: Force ' . {
+        cddvd => 'CD/DVD Boot',
+        hdd => 'Hard-drive Boot',
+        pxe => 'PXE Boot',
+    }->{$bootdev};
```

Comment:
```suggestion
    }->{$bootdev} or die "Unsupported boot device $bootdev";
```

Reply:
Good idea, but that must be in parens because of precedence:
```
my $amt_bootdev = 'Intel(r) AMT: Force ' . ({
...
}->{$bootdev} or die "Unsupported boot device $bootdev");
```

Reply:
I guess ` .. || or die "...";` is a common style. Did that now.

Reply:
That won't help :)
The `"string" . { ... }->{$key}` concatenation is the problem.
```
my $key = "a";
my $str = "foo " . { 
  x => 1, y => 2, z => 3
}->{$key} || die "nope"; 
say "str: $str";
__END__
Use of uninitialized value in concatenation (.) or string at -e line 5.
str: foo 
```

Reply:
And apart from that, `or die` is the more common style, and `|| die` can be wrong sometimes, e.g.
```
open my $fh, "/tmp/nonsense" || die "Could not open";
```
this won't die.

Reply:
Ok, I thought || is more common. I will change

Reply:
done